ui/app/pages/swaps/dropdown-search-list/dropdown-search-list.js

```diff
@@ -29,12 +29,15 @@ export default function DropdownSearchList ({
   const t = useContext(I18nContext)
   const [isOpen, setIsOpen] = useState(false)
   const [selectedItem, setSelectedItem] = useState(startingItem)
+  const close = () => {
+    setIsOpen(false)
+    onClose && onClose()
+  }
   const onClickItem = useCallback((item) => {
     onSelect && onSelect(item)
     setSelectedItem(item)
-    setIsOpen(false)
-    onClose && onClose()
-  }, [onClose, onSelect])
+    close()
+  }, [onSelect])
```

Comment:
```suggestion
  }, [close, onSelect])
```

without close being in the dependency array, `useCallback` would only get called anytime onSelect changes, which could result in holding onto a reference to the close function from previous renders (and also anything that also uses that reference to the close function).